docs/content/saml: add Azure AD section (#4136)
chef/automate: components/automate-chef-io/content/docs/saml.md

## Supported Identity Management Systems

- [Azure AD]({{< relref "#azure-ad" >}})
- Office365
- OKTA
- OneLogin
- Ping
- Tivoli Federated Identity Manager

In the following section, you can find further details about specific supported identity management systems.

### Azure AD

When using Chef Automate's SAML integration with Azure AD, please note that Chef Automate does not support automatic **signing key rotation**. The signing certificate used with Azure for SAML sign-ins has to be **managed manually**.

After setting up your application as a _"non-gallery application"_, configure its SAML sign-on method.
For both _Identifier (Entity ID)_ and _Reply URL (Assertion Consumer Service URL)_, enter
`https://{{< example_fqdn "automate" >}}/dex/callback`.

The default claims can be used, see below for their counterparts in the Chef Automate configuration.
Download the _Certificate (Base64)_, and take note of the _Login URL_.

The minimal configuration snippet then looks like this (replace `<<something>>` with the values
noted above):

```toml
[dex.v1.sys.connectors.saml]
ca_contents="""
<<Certificate (Base64)>>
"""

> sso_url = "<< Login URL >>"

Reviewer: Should be consistent: either spaces just inside the angle brackets (L73) or not (L71).

## Supported Identity Management Systems

- [Azure AD]({{< relref "#azure-ad" >}})
- Office365
- OKTA
- OneLogin
- Ping
- Tivoli Federated Identity Manager

In the following section, you can find further details about specific supported identity management systems.

### Azure AD

> For using Chef Automate's SAML integration with Azure AD, please note that **signing key rotation** is
> not supported to be done automatically. The signing certificate used with Azure for SAML sign-ins
> has to be **managed manually**.

Reviewer 2: Maybe something like: 

```suggestion
When using Chef Automate's SAML integration with Azure AD, please note that Chef Automate does not support automatic **signing key rotation**. The signing certificate used with Azure for SAML sign-ins has to be **managed manually**.
```

Author: Taken, thanks!

> After setting up your application as a _"non-gallery application"_, configure its SAML sign-on method.

Reviewer 3: Are the fields in _italics_ supposed to be the relevant fields for the Azure AD set-up (and not directly for the Chef Automate interface)?

Author: Yeah! Is there another markup you'd like to have instead?

Reviewer 3: Italics are fine. I just wanted to make sure that I was interpreting that those italics were for fields/labels since I can't reference them for myself. Thanks!

For both _Identifier (Entity ID)_ and _Reply URL (Assertion Consumer Service URL)_, enter

> `https://{{< example_fqdn "automate" >}}/dex/callback`.

Reviewer 2: Do we want to guide them to _where_ they should be "setting up your application" or is this obvious to someone using Azure AD?

Author: I'm not sure. What do you think? 👇 

![Screenshot 2020-07-27 at 15 30 39](https://user-images.githubusercontent.com/870638/88548286-33cf2b00-d01f-11ea-806b-35887a622e52.png)

Here, we select "View" below the middle tile, and get to

![Screenshot 2020-07-27 at 15 32 09](https://user-images.githubusercontent.com/870638/88548381-506b6300-d01f-11ea-95c8-c731cfe03cb5.png)

...where clicking on "Enterprise application" will get us to

![Screenshot 2020-07-27 at 15 32 15](https://user-images.githubusercontent.com/870638/88548471-6ed15e80-d01f-11ea-8648-868f5eae460f.png)

where we find the "non-gallery application".

I'm sure there's other ways. I think I'd like to avoid adding too much of that here, since then our docs are tied to a UI in Azure; which may change at some point.

Reviewer 3: Would this piece of Azure documentation be relevant and a possible link to include? https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/add-application-portal#add-an-app-to-your-azure-ad-tenant

Author: Hmm I don't know about that specific link, it seems to be a little misleading: it's describing the github integration, which is easier to set up.

Do you prefer having some official docs to link to? I can try digging up a more fitting link.

Reviewer 3: I know we won't find specific docs referring to Chef, but it would be good to link official docs that covers the general aspects of Azure's "adding an application".  That way, our Chef Automate documentation can provide the needed details rather than having to recreate the wheel with detailed documentation that may change.

Author: Yeah we're on the same page here. I'll go look for a generic docs page that we can reference. :)

The default claims can be used, see below for their counterparts in the Chef Automate configuration.
Download the _Certificate (Base64)_, and take note of the _Login URL_.

> The minimal configuration snippet then looks like this (replace `<<something>>` with the values

Reviewer 3: Is this the minimal configuration snippet for Chef Automate? If so, what file does this example refer to?

Author: It's the minimal SAML-related config to make this work, when patching it into the running settings.

## Supported Identity Management Systems

- [Azure AD]({{< relref "#azure-ad" >}})
- Office365
- OKTA
- OneLogin
- Ping
- Tivoli Federated Identity Manager

### Azure AD

Using Azure AD as an SAML IdP requires specific configuration for both Azure AD and Chef Automate.

{{< info >}}
The signing certificate used for Chef Automate's SAML integration with Azure AD requires manual management.
Signing key rotation is not done automatically.
{{< /info >}}

> First, set up your configuration within Azure AD.

Reviewer 3: @srenatus I would think we could put your official documentation link towards the beginning on either line 61 or 62? Either line may have to be reworded. I just realized both sentences begin with "set up."

Author: Linked something :) Not sure about the wording there, though -- please have a look.

Set up your application as a _"non-gallery application"_, and then configure its SAML log-in method.

> Enter `https://{{< example_fqdn "automate" >}}/dex/callback` as the value for both _Identifier (Entity ID)_ and _Reply URL (Assertion Consumer Service URL)_,.

Reviewer 3: @srenatus Ideally, I would like to have a bulleted list with the Azure AD field & its expected value listed. Since I assume these on line 63 are the only two required fields and respective value, I wasn't going to change the format here.

Author: From playing with it, these are sufficient. Besides managing the certificate.

The original CFT docs (linked in https://github.com/chef/automate/issues/2887) included renaming the claims, but I've found that to be not necessary.

Reviewer 4: The only thing I would add would to include `name_id_policy_format = "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"` in the saml.toml. If I remember correctly, I had to do this to get Automate to use the email address as a login instead of the user UUID from AAD.

Author: Hrm I've tried it and it worked without. So, I'd rather keep this as simple as can be...? 🤔

Reviewer 4: Cool. I set mine lab up several months ago so it's very possible some other thing got fixed between now and then. I'm good leaving it as simple as possible.

- Ping
- Tivoli Federated Identity Manager

### Azure AD

Using Azure AD as an SAML IdP requires specific configuration for both Azure AD and Chef Automate.

{{< info >}}
The signing certificate used for Chef Automate's SAML integration with Azure AD requires manual management.
Signing key rotation is not done automatically.
{{< /info >}}

First, set up your configuration within Azure AD.

> Set up your application as a _"non-gallery application"_, and then configure its SAML log-in method.

Author: I believe we've been trying to standardize on "sign-in" instead of "log-in" -- am I on the wrong track here?
